posts/2024-05-07-24.0.0.5-beta.adoc

```diff
@@ -0,0 +1,554 @@
+
+To use the CDI 4.1 feature, add `cdi-4.1` to your `server.xml` file:
+
+[source,xml]
+----
+<featureManager>
+  <feature>cdi-4.1</feature>
+</featureManager>
+----
+
+Further references:
+
+* link:https://jakarta.ee/specifications/cdi/4.1/apidocs/[CDI API]
+* link:https://jakarta.ee/specifications/interceptors/2.2/apidocs/[Interceptors API]
+* link:https://jakarta.ee/specifications/annotations/3.0/apidocs/jakarta.annotation/module-summary.html[Jakarta Annotations API]
+* link:https://jakarta.ee/specifications/cdi/4.1/jakarta-cdi-spec-4.1[CDI Specification]
+* link:https://jakarta.ee/specifications/interceptors/2.2/jakarta-interceptors-spec-2.2[Interceptors Specification]
+* link:https://jakarta.ee/specifications/annotations/3.0/annotations-spec-3.0[Jakarta Annotations Specification]
+
+// DO NOT MODIFY THIS LINE. </GHA-BLOG-TOPIC>
+
+
+// // // // DO NOT MODIFY THIS COMMENT BLOCK <GHA-BLOG-TOPIC> // // // //
+// Blog issue: https://github.com/OpenLiberty/open-liberty/issues/28075
+// Contact/Reviewer: mswatosh
+// // // // // // // //
+[#concurrent]
+== Jakarta Concurrency 3.1
+
+Jakarta Concurrency allows applications to use concurrency while maintaining the benefits of running on a Jakarta EE runtime. Jakarta Concurrency 3.1 is the new version for the upcoming Jakarta EE 11, and it adds support for new features of the latest Java SE releases as well as some usability improvements.
+
+The new features in Jakarta Concurrency 3.1 are improved support for the Java Flow (Reactive Streams) APIs, and the use of virtual threads in `ManagedExecutors` and `ManagedThreadFactories`. Support is also included for using `@Inject` in place of `@Resource`, and the option to run asynchronous methods on a schedule by using the new `@Schedule` annotation.
+
+To support the Flow APIs, `ContextService` has two new methods, `contextualSubscriber`, which provides context in a `Flow.Subscriber`, and `Flow.Processor`, which provides context in a `contextualProcessor`. These new methods allow for context in a flow when using a `Flow.Publisher` which does not allow configuring a `ManagedExecutor`.
+
+[source,java]
+----
+publisher.subscribe(contextService.contextualSubscriber(subscriber));
+----
+
+Virtual threads are available in Jakarta Concurrency 3.1 by specifying ‘virtual = true’ on a `ManagedExecutorDefinition`, `ManagedScheduledExecutorDefinition`, or `ManagedThreadFactory` instance. Virtual threads are a new feature of Java 21, so when running on Java 17 if ‘virtual = true’ is specified, platform threads are provided instead of virtual threads.

```

Comment:
I think it reads better like this. Feel free to ignore the comment if you don't agree.
```suggestion
Virtual threads are available in Jakarta Concurrency 3.1 by specifying ‘virtual = true’ on a `ManagedExecutorDefinition`, `ManagedScheduledExecutorDefinition`, or `ManagedThreadFactory` instance. Virtual threads are a new feature of Java 21, so when running on Java 17 and ‘virtual = true’ is specified, platform threads are provided instead of virtual threads.

```